lib/wallet/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -1146,13 +1146,14 @@ binaryCalculationsSpec' era = describe ("calculateBinary - "+||era||+"") $ do
     calculateBinary net utxo outs chgs pairs =
         hex (Cardano.serialiseToCBOR ledgerTx)
       where
+          ledgerTx :: Cardano.Tx (CardanoApiEra era)
```

Comment:
`Cannot satisfy: cardano-ledger-binary-1.2.1.0:Cardano.Ledger.Binary.Version.MinVersion <= Ledger.ProtVerHigh era0` without type annotation 🤔 maybe we should have kept the `era` arg here then?

Reply:
This was the error I saw when testing dropping `era` from `constructTransaction` also 🤔

Reply:
In the presence of GADTs, it is good practice — and sometimes necessary — to annotate polymorphic types in `where` or `let` bindings.

More information can be found in the documentation for the [`MonoLocalBinds`](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/let_generalisation.html) extensions, which is implied by the `GADTs` extension.

In other words: Using `GADTs` has a cost. `MonoLocalBinds` is that cost. We pay it by sometimes adding type signatures.